commands/command_push.go

```diff
@@ -45,7 +46,7 @@ func uploadsBetweenRefAndRemote(remote string, refs []string) *lfs.TransferQueue
 		if len(refs) == 0 {
 			pointers := scanAll()
 			Print("Pushing objects...")
-			return uploadPointers(pointers)
+			return uploadPointers(pointers, strings.Join(refs, ","))
```

Comment:
Since `pre-push` is reading from the git hook, it's probably getting a full remote ref like `refs/heads/master`. However, the `push` command is receiving user input, and is likely a relative name (which could be a branch, tag, etc). Do we care? Should the server just do its best to validate it?

Also, instead of joining the refs array with commas, why not just pass a JSON array to the server.